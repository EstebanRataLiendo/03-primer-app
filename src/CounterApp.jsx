//importo los states
import { useState } from 'react';
//importo los props
import PropTypes from 'prop-types';

//creo y exporto una función que usa states
export const CounterApp = ({value})=> {

    //desestructuro dos valores, el primer elemento tendrá el valor inicial del state que le demos
    //luego se irá actualizando según los hooks
    //el segundo elemento es el que le dice a React que hubo un cambio en el state y debe actualizarse
    const [ counter, setCounter ] = useState( value )

    //creo una función que agrega 1 al primer elemento que desestructuré y se lo asigno al segundo elemento
    const sumar= ()=> setCounter(counter+1)
    const restar= ()=> setCounter(counter-1)
    const reset= ()=> setCounter(value)

    //actualizo la pagina con jsx
    return (
    <>
        <h1>CounterApp</h1>
        <h2> {counter} </h2>
        <button onClick={sumar}>+1</button>
        <button aria-label="btn-1" onClick={restar}>-1</button>
        <button onClick={reset}>Reset</button>
    </>
    )
}

//le digo a la app que el elemento value tiene que ser numerico y es requerido 
// CounterApp.propTypes= {
//     value: PropTypes.number.isRequired
// }
