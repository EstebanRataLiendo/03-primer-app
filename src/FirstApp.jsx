import PropTypes from 'prop-types';


export const FirstApp = ({title ='Titulo Default',subTitle='Subtitulo Default'}) => {
    return (
    <>
        {/* se puede cargar expresiones pero no un objeto */}
        <h1 data-testid='test-title'>{title}</h1>
        {/* <h2>{newMessage}</h2>
        <h4> {nMessage.title} </h4>
        <h5> {100*100+' '+ 'String desde JSX con operaciones'} </h5>
        <code> {JSON.stringify (nMessage)} </code>
        <h3> {neMessage} </h3> */}
        <p>{subTitle}</p>

    </>
    )
}

FirstApp.propTypes= {
    title: PropTypes.string,
    subTitle: PropTypes.string//.isRequired
}

