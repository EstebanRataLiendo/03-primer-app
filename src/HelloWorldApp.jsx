export const App= ()=>{
    //document.createElement...
    return <h1>String desde funciÃ³n JSX</h1>
};



