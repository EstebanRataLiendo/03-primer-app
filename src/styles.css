html,body{
    background-color: #21232B;
    color: white;
    font-family: Arial, Helvetica, sans-serif;
    font-size: 1.3rem;
    padding: 70px;
}
button{
    padding: 15px;
    font-size: 1.3rem;
    color: lightgrey;
    background-color: darkblue;
    border-radius: 10%;
    margin-right: 5px;
}