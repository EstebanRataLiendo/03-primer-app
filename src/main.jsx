import React from 'react'
import ReactDom from 'react-dom/client';
import { CounterApp} from './CounterApp'


import {App} from './HelloWorldApp'
import { FirstApp } from './FirstApp';

//importar estilo
import './styles.css'


ReactDom.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        {/* <App />
        <FirstApp title='String desde props del main(se esperaba Numero)' /> */}
        <CounterApp value={100} />
        {/* <FirstApp subTitle='Subtitulo definido de antemano'/> */}
    </React.StrictMode>
)