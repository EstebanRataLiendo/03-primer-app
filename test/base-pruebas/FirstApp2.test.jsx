import { render,screen } from "@testing-library/react"
import { FirstApp } from "../../src/FirstApp"

describe('Pruebas en FirstApp', () => {
    const title = 'Hola mundo!';
    const subTitle = 'Hola subtitulo!';

    test('debe hacer match con el snapshot', () => {
        const {container}= render (<FirstApp title={title} />);
        expect (container).toMatchSnapshot();
    })
    
    test('debe mostrar el mensaje "Hola Mundo!!!" ', () => {
        render (<FirstApp title={title} />);
        expect(screen.getByText(title)).toBeTruthy();
    })
    
    test('debe mostrar el titulo en h1', () => {
        render (<FirstApp title={title} />);
        expect(screen.getByRole('heading', {level:1}).innerHTML).toContain(title);
    })
    
    test('debe mostrar el subtitulo enviado por props', () => {
        render (
            <FirstApp 
                title={title}
                subTitle={subTitle}
            />);
        expect(screen.getAllByText(subTitle).length).toBe(1);
    })
})

