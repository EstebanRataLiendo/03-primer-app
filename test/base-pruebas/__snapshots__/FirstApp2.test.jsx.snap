// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola mundo!
  </h1>
  <p>
    Subtitulo Default
  </p>
</div>
`;
